verifcid/validate.go

```diff
@@ -33,37 +33,15 @@ var goodset = map[uint64]bool{
 	mh.SHA1: true, // not really secure but still useful
 }
 
+// IsGoodHash checks for multihash allowance by the given code.
 func IsGoodHash(code uint64) bool {
-	good, found := goodset[code]
-	if good {
-		return true
-	}
-
-	if !found {
-		if code >= mh.BLAKE2B_MIN+19 && code <= mh.BLAKE2B_MAX {
-			return true
-		}
-		if code >= mh.BLAKE2S_MIN+19 && code <= mh.BLAKE2S_MAX {
-			return true
-		}
-	}
-
-	return false
+	return DefaultAllowlist.IsAllowed(code)
 }
 
+// ValidateCid validates multihash allowance behind given CID.
 func ValidateCid(c cid.Cid) error {
-	pref := c.Prefix()
-	if !IsGoodHash(pref.MhType) {
-		return ErrPossiblyInsecureHashFunction
-	}
-
-	if pref.MhType != mh.IDENTITY && pref.MhLength < minimumHashLength {
-		return ErrBelowMinimumHashLength
-	}
-
-	if pref.MhType != mh.IDENTITY && pref.MhLength > maximumHashLength {
-		return ErrAboveMaximumHashLength
-	}
-
-	return nil
+	return DefaultAllowlist.ValidateCid(c)
 }
+
+// DefaultAllowlist is the default list of hashes allowed in IPFS.
+var DefaultAllowlist = NewAllowlist(goodset)
```

Comment:
Having a `DefaultAllowlist` is good, but I think we should take advantage of it and use a stricter implementation.
Something like `binary.LittleEndian` would be better and ward off allocations by having a concrete type.

So:
```suggestion
var DefaultAllowlist defaultAllowlist
```
And then defining `IsAllowed` to use a big switch statement.

Reply:
I don't mind following the `binary` pkg here

Reply:
@Jorropo, but what do you mean by?
> And then defining IsAllowed to use a big switch statement.

We have goodset map, not a switch statement.

Reply:
> I don't mind following the binary pkg here

Actually, if we do so, we will lose the default implementation of Allowlist and if we still want to keep it, then we will have to repeat the code: in default and public impls.

Reply:
> Actually, if we do so, we will lose the default implementation of Allowlist and if we still want to keep it, then we will have to repeat the code: in default and public impls.

We can keep it and just not use it here.

I don't see what code would be repeated.

Reply:
Ok, your idea to eliminate ValidateCID from the interface resolves my concern!